### PR TITLE
Ofir/add stripe service

### PR DESCRIPTION
**Pull request by OfirNadav**

### CODE REVIEW

File: demo/stripe/demo.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        .decorator('AuthService', function ($delegate) {
+            'ngInject';
+            $delegate.getAuth = function () {
+                return new Promise(resolve => resolve({uid: 'cf656635-b44c-4fcc-82ef-72fe566d5540'}));
+            };
+            return $delegate;
+        })
+        .value('ENV',  {
+            znkBackendBaseUrl: 'https://dev-api.zinkerz.com',
+            stripeToken: 'pk_test_VCX5a5aw64Z8WbB811quSQMj'
```
Comment (assafshp) [inline demo/stripe/demo.js]: a bit issue since it is a public repo....although in view source every user can see it.

File: demo/stripe/demo.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        })
+        .controller('Main', function ($scope, StripeService) {
+            $scope.openPopup = () => {
+                const amount = '20';
+                const name = 'Zinkerz';
+                const description = 'Instance Payment Description';
+                const image = 'stripe/assets/images/zinkerz_stripe_logo.jpg';
+                StripeService.openStripeModal(amount, name, description, image)
```
Comment (assafshp) [inline demo/stripe/demo.js]: In paypal we have:
- uid
- product id
- firebase url 
- app
an example:
AMIrLVjlBIZpJm7nmtXjLzdok7E2#com.zinkerz.act.allcontent#https://act-dev.firebaseio.com/#act_app

I think we should be the same

File: dist/stripe/stripe.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    'use strict';
+
+    angular.module('znk.infra-web-app.stripe')
+        .service('StripeService',
+            ["$log", "$q", "$window", "$filter", "$http", "AuthService", "ENV", function ($log, $q, $window, $filter, $http, AuthService, ENV) {
+                'ngInject';
+
+                const stripeToken = ENV.stripeToken;
+                const translate = $filter('translate');
+                const paymentApi = `${ENV.znkBackendBaseUrl}/payment`;
```
Comment (assafshp) [inline dist/stripe/stripe.js]: are you sure ENV.znkBackendBaseUrl is the new api and not the old one ?

File: dist/stripe/stripe.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                 * @returns StripeResult object{Subject<boolean>}
+                 *
+                 * stripe test card 4242 4242 4242 4242
+                 */
+                this.openStripeModal = (amount, name, description, image) => {
+                    let tokenId = null;
+                    const defer = $q.defer();
+                    const handler = ($window).StripeCheckout.configure({
+                        key: stripeToken,
+                        amount: amount ? amount * 100 : 0, // amount to display: ex: $20 * 100 === $20.00
```
Comment (assafshp) [inline dist/stripe/stripe.js]: I think you should have another var amountIn cents....

File: dist/stripe/stripe.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                        locale: 'auto',
+                        token: token => tokenId = token.id
```
Comment (assafshp) [inline dist/stripe/stripe.js]: what if you didn't get a token ?

File: dist/stripe/stripe.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                    });
+                    handler.open({
+                        name: name || 'Zinkerz',
```
Comment (assafshp) [inline dist/stripe/stripe.js]: Isn't it should come from firebase iap product ?

File: dist/stripe/stripe.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+                        description: description || translate('STRIPE.DESCRIPTION'),
```
Comment (assafshp) [inline dist/stripe/stripe.js]: I think it should also come from product in firebase

File: dist/stripe/stripe.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+                    return resToReturn;
+                }
+
+                function createInstantCharge(tokenId, amount, description) {
+                    const createInstantChargeApi = `${paymentApi}/createinstantcharge`;
+                    return AuthService.getAuth().then(authData => {
+                        const stripeVars = {
+                            token: tokenId,
+                            uid: authData.uid,
```
Comment (assafshp) [inline dist/stripe/stripe.js]: maybe check that you do have a uid....

File: dist/stripe/stripe.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+                            amount: amount,
+                            description: description,
+                            currency: 'usd'
```
Comment (assafshp) [inline dist/stripe/stripe.js]: 'usd' -> should be const

Comment (assafshp) [reply dist/stripe/stripe.js]: in paypal we have the service (=app) and the firebase url....don't we need it here also ?

File: demo/stripe/demo.js (hunk 1 above)
Comment (OfirNadav) [reply demo/stripe/demo.js]: OK

File: demo/stripe/demo.js (hunk 2 above)
Comment (OfirNadav) [reply demo/stripe/demo.js]: product id will be added to next version, let's talk about it

File: dist/stripe/stripe.js (hunk 3 above)
Comment (OfirNadav) [reply dist/stripe/stripe.js]: yes

File: dist/stripe/stripe.js (hunk 4 above)
Comment (OfirNadav) [reply dist/stripe/stripe.js]: fixed

File: dist/stripe/stripe.js (hunk 5 above)
Comment (OfirNadav) [reply dist/stripe/stripe.js]: then it goes to closed Fn and nothing happen

File: dist/stripe/stripe.js (hunk 6 above)
Comment (OfirNadav) [reply dist/stripe/stripe.js]: This is a generic stripe modal.
I don't have the serviceId or the productId so we don't know at this point which product is this.
this Fn just return the token the payment with all of the params done in the backend

File: dist/stripe/stripe.js (hunk 7 above)
Comment (OfirNadav) [reply dist/stripe/stripe.js]: see answer above

File: dist/stripe/stripe.js (hunk 8 above)
Comment (OfirNadav) [reply dist/stripe/stripe.js]: fixed

**Pull request merged**